tests/queries/0_stateless/02473_optimize_old_parts.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+# Tags: long
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+# Wait for number of parts in table $1 to change to anything other than "$2".
```

Comment:
```suggestion
# Wait for number of parts in table $1 to change to anything other than "$2".
```

you need to wait the exact number because, not just the change.
If we don't merge the entire partition automatically we can first merge 2 old parts and have 2 instead of 1 part.